src/plugins/intel_gpu/tests/unit/dynamic_execution/skip_redundant_reorder_at_runtime.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "test_utils.h"
+
+#include <intel_gpu/primitives/input_layout.hpp>
+#include <intel_gpu/primitives/reorder.hpp>
+#include <intel_gpu/primitives/data.hpp>
+#include <intel_gpu/primitives/fully_connected.hpp>
+
+#include "program_wrapper.h"
+
+#include <cmath>
+#include <algorithm>
+
+using namespace cldnn;
+using namespace ::tests;
+
+namespace skip_reorder_tests {
+TEST(remove_redundant_reorder, skip_reorder_at_runtime) {
+    auto& engine = get_test_engine();
+
+    auto weight_mem = engine.allocate_memory({{2, 32}, data_types::f32, format::bfyx});
+    std::vector<float> weight_data(weight_mem->get_layout().count());
+    std::iota(weight_data.begin(), weight_data.end(), 1.0f);
+    set_values(weight_mem, weight_data);
+
+    auto input_l = layout{ov::PartialShape::dynamic(2), data_types::f32, format::bfyx};
+    topology topology(input_layout("input", input_l),
+                      data("weight", weight_mem),
+                      fully_connected("fc", input_info("input"), {"weight"}, "", data_types::f32),
+                      reorder("reorder", input_info("fc"), format::bfyx, data_types::f32)); /*output padding*/
```

Comment:
How about adding another reorder which is not output? I suggest this because output reorder is sometimes handled as specific case.

Reply:
Skipping output reorder is my intention for dynamic case.